### Inferences About Population variances 

The Fidelity Growth & Income mutual fund received a three-star, or neutral, rating from Morningstar. 
Shown here are the quarterly percentage returns for the five-year period from 2001 to 2005 (Morningstar Funds 500, 2006).

a. Compute the mean, variance, and standard deviation for the quarterly returns.

b. Financial analysts often use standard deviation as a measure of risk for stocks and mutual funds. Develop a 95% confidence interval for the population standard deviation
of quarterly returns for the Fidelity Growth & Income mutual fund.

In [35]:
import pandas as pd
import numpy as np
from scipy import stats
Financial_Data = pd.read_csv("D:\Data for Preprocessing\Financial Data .csv", index_col=0)
Financial_Data

1st Quarter   2nd Quarter   3rd Quarter  4th Quarter
2001        -10.91          5.80        -9.64         6.45
2002          0.83        -10.48       -14.03         5.58
2003         -2.27         10.43         0.85         9.33
2004          1.34          1.11        -0.77         8.03
2005         -2.46          0.89         2.55         1.78

In [36]:
# Calculate mean, variance, and standard deviation for the quartely returns
mean = Financial_Data.mean() 
variance = Financial_Data.var(ddof = 1)  # set ddof = 1 for sample variance
standard_deviation = Financial_Data.std(ddof=1)# set ddof = 1 for standard deviation

# Display the results 
print("Mean Quarterly Returns:")
print(mean)

print("\nVariance of Quarterly Returns:")
print(variance)

print("\nStandard Deviation of Quarterly Returns:")
print(standard_deviation)

# Calculate the 95% confidence interval for the population standard deviation
confidence_level = 0.95
sample_size = Financial_Data.shape[0] 
degree_of_freedom = sample_size - 1  # Degree of Freedom for sample variance

# use the chi-square distribution to calculate the confidence interval
lower_bound = np.sqrt((degree_of_freedom * variance) / stats.chi2.ppf((1 + confidence_level) / 2, df = degree_of_freedom))
upper_bound = np.sqrt((degree_of_freedom * variance) / stats.chi2.ppf((1 - confidence_level) / 2, df = degree_of_freedom))

# Display the 95% confidence interval for the population standard deviation 
print("\n95% Confidence Interval for Population Standard Deviation")
print("Lower Bound:", lower_bound)
print("Upper Bound:", upper_bound)

Mean Quarterly Returns:
1st Quarter    -2.694
2nd Quarter     1.550
3rd Quarter    -4.208
4th Quarter     6.234
dtype: float64

Variance of Quarterly Returns:
1st Quarter     24.10723
2nd Quarter     60.56675
3rd Quarter     52.26302
4th Quarter      8.28083
dtype: float64

Standard Deviation of Quarterly Returns:
1st Quarter     4.909911
2nd Quarter     7.782464
3rd Quarter     7.229317
4th Quarter     2.877643
dtype: float64

95% Confidence Interval for Population Standard Deviation
Lower Bound: 1st Quarter     2.941691
2nd Quarter     4.662732
3rd Quarter     4.331323
4th Quarter     1.724091
dtype: float64
Upper Bound: 1st Quarter     14.108904
2nd Quarter     22.363344
3rd Quarter     20.773844
4th Quarter      8.269068
dtype: float64


    Ans (a) Mean, variance, and standard deviation for the quartely returns
    
            Mean Quarterly Returns:
            1st Quarter    -2.694
            2nd Quarter     1.550
            3rd Quarter    -4.208
            4th Quarter     6.234

            Variance of Quarterly Returns:
            1st Quarter     24.10723
            2nd Quarter     60.56675
            3rd Quarter     52.26302
            4th Quarter      8.28083

            Standard Deviation of Quarterly Returns:
            1st Quarter     4.909911
            2nd Quarter     7.782464
            3rd Quarter     7.229317
            4th Quarter     2.877643

    Ans (b) 95% Confidence Interval for Population Standard Deviation

            Lower Bound:    1st Quarter     2.941691
                            2nd Quarter     4.662732
                            3rd Quarter     4.331323
                            4th Quarter     1.724091
            
            Upper Bound:    1st Quarter     14.108904
                            2nd Quarter     22.363344
                            3rd Quarter     20.773844
                            4th Quarter      8.269068
            

### Inferences About Two Population variances 

Media Metrix and Jupiter Communications gathered data on the time adults and the time teens spend online during a month (USA Today, September 14, 2000). The study concluded that on average, adults spend more time online than teens. Assume that a follow-up study sampled 26 adults and 30 teens. The standard deviations of the time online during a month were 94 minutes and 58 minutes, respectively. 

Do the sample results support the conclusion that adults have a greater variance in online time than teens? Use α = .01. What is the p-value?

    Ans :
         Null Hypothesis (H0): The variance of online time for adults is equal to or less than the variance of online time for teens.
         
         Alternative Hypothesis (H1): The variance of online time for adults is greater than the variance of online time for teens.

In [39]:
# Sample data
n_adults = 26
n_teens = 30

variance_adults = 94**2
variance_teens = 58**2

# Calculating the F-statistic
F_statistic = variance_adults/variance_teens

# Degree of freedom
df1 = n_adults - 1
df2 = n_teens - 1

# Calculating the p-value
p_value = 1- stats.f.cdf(F_statistic, df1, df2)

print("F_statistic:", F_statistic)
print("P_value:", p_value)

F_statistic: 2.6266349583828776
P_value: 0.006733349965066915


    To conclude, 
    p-value is less than α (0.01 in this case), we reject the null hypothesis in favor of the alternative hypothesis, which means there is evidence that adults 
    have a greater variance in online time than teens.